tests/platform/thermal_control_test_helper.py

```diff
@@ -0,0 +1,367 @@
+    find_thermalctld_pid_cmd = 'docker exec -i pmon bash -c \'pgrep thermalctld | sort\''
+    output = dut.command(find_thermalctld_pid_cmd)
+    assert output["rc"] == 0, "Run command '%s' failed" % find_thermalctld_pid_cmd
+    assert len(output["stdout_lines"]) == 2, "There should be 2 thermalctld process"
+    pid_0 = int(output["stdout_lines"][0].strip())
+    pid_1 = int(output["stdout_lines"][1].strip())
+    # find and kill the parent process
+    pid_to_kill = pid_0 if pid_0 < pid_1 else pid_1
+    logging.info('Killing old thermal control daemon with pid: {}'.format(pid_to_kill))
+    kill_thermalctld_cmd = 'docker exec -i pmon bash -c \'kill {}\''.format(pid_to_kill)
+    output = dut.command(kill_thermalctld_cmd)  # kill thermalctld and wait supervisord auto reboot thermalctld
+    assert output["rc"] == 0, "Run command '%s' failed" % kill_thermalctld_cmd
+    
+    # make sure thermalctld has restarted
+    max_wait_time = 5
+    while max_wait_time > 0:
+        max_wait_time -= 1
+        output = dut.command(find_thermalctld_pid_cmd)
+        assert output["rc"] == 0, "Run command '%s' failed" % find_thermalctld_pid_cmd
+        if len(output["stdout_lines"]) != 2:
+            time.sleep(1)
+            continue
+        
+        new_pid_0 = int(output["stdout_lines"][0].strip())
+        new_pid_1 = int(output["stdout_lines"][1].strip())
+        parent_pid = new_pid_0 if new_pid_0 < new_pid_1 else new_pid_1
+        
+        if parent_pid == pid_to_kill:
+            logging.info('Old thermal control daemon is still alive, waiting...')
+            time.sleep(1)
+            continue
+        else:
+            logging.info('New pid of thermal control daemon is {}'.format(parent_pid))
+            return
+
+    assert 0, 'Wait thermal control daemon restart failed'
+
+
+class ThermalPolicyFileContext:
+    """
+    Context class to help replace thermal control policy file and restore it automatically.
+    """
+    def __init__(self, dut, src):
+        """
+        Constructor of ThermalPolicyFileContext.
+        :param dut: DUT object representing a SONiC switch under test.
+        :param src: Local policy file path.
+        """
+        self.dut = dut
+        self.src = src
+        platform_str = dut.facts['platform']
+        self.thermal_policy_file_path = DUT_THERMAL_POLICY_FILE.format(platform_str)
+        self.thermal_policy_file_backup_path = DUT_THERMAL_POLICY_BACKUP_FILE.format(platform_str)
+
+    def __enter__(self):
+        """
+        Back up original thermal control policy file and replace it with the given one. Restart
+        thermal control daemon to make it effect.
+        :return:
+        """
+        self.dut.command('mv -f {} {}'.format(self.thermal_policy_file_path, self.thermal_policy_file_backup_path))
+        self.dut.copy(src=os.path.join(FILES_DIR, self.src), dest=self.thermal_policy_file_path)
+        restart_thermal_control_daemon(self.dut)
```

Comment:
should consider the situation that thermal control daemon restart fails. same comments apply to the below exit function.

Reply:
fixed